ament_pep257/ament_pep257/main.py

```diff
@@ -187,7 +187,7 @@ def generate_pep257_report(paths, excludes, ignore, select, convention, add_igno
     report = []
 
     files_dict = {}
-    for filename, checked_codes, ignore_decorators in files_to_check:
+    for filename, checked_codes, ignore_decorators, _ in files_to_check:
```

Comment:
So this isn't going to work with older pydocstyle, which we still need to support for users who haven't upgraded the pydocstyle yet.

Instead, I think we'll need to first try to unpack 3 values, and if that fails with a `ValueError`, then try to unpack with 4 values.  Alternatively, we could look up the version number of pydocstyle that is currently in use, and choose which one to do based on that (though I think that is a little more brittle).